x/rollup/keeper/deposits.go

```diff
@@ -85,19 +94,85 @@ func (k *Keeper) processL1UserDepositTxs(ctx sdk.Context, txs [][]byte) (sdk.Eve
 			return nil, types.WrapError(types.ErrMintETH, "failed to mint ETH for cosmosAddress: %v; err: %v", cosmAddr, err)
 		}
 		mintEvents = append(mintEvents, *mintEvent)
+
+		// TODO: remove logic to get the from address once https://github.com/polymerdao/monomer/pull/208 is merged
+		from, err := ethtypes.NewCancunSigner(tx.ChainId()).Sender(&tx)
+		if err != nil {
+			ctx.Logger().Error("Failed to get sender from tx", "index", i, "err", err)
+			return nil, types.WrapError(types.ErrInvalidL1Txs, "failed to get sender from tx, index:%d, err:%v", i, err)
+		}
+
+		// TODO: remove hardcoded address once a genesis state is configured
+		// Convert the L1CrossDomainMessenger address to its L2 aliased address
+		aliasedL1CrossDomainMessengerAddress := crossdomain.ApplyL1ToL2Alias(common.HexToAddress("0x9A9f2CCfdE556A7E9Ff0848998Aa4a0CFD8863AE"))
+
+		// Check if the tx is a cross domain message from the aliased L1CrossDomainMessenger address
+		if from == aliasedL1CrossDomainMessengerAddress && tx.Data() != nil {
+			erc20mintEvent, err := k.parseAndExecuteCrossDomainMessage(ctx, tx.Data())
+			// Do not return an error if the cross domain message can't be parsed or executed to keep the
+			// state transitions from the other deposit txs
+			if err != nil {
+				// TODO: should this log be a warning or error?
+				ctx.Logger().Warn("Failed to parse or execute cross domain message", "err", err)
+			} else if erc20mintEvent != nil {
+				mintEvents = append(mintEvents, *erc20mintEvent)
+			}
+		}
 	}
 
 	return mintEvents, nil
 }
 
+// parseAndExecuteCrossDomainMessage parses the tx data of a cross domain message and applies state transitions for recognized messages.
+// Currently, only finalizeBridgeERC20 messages from the L1StandardBridge are recognized for minting ERC-20 tokens on the Cosmos chain.
+// If a message is not recognized, it returns nil and does not error.
+func (k *Keeper) parseAndExecuteCrossDomainMessage(ctx sdk.Context, txData []byte) (*sdk.Event, error) { //nolint:gocritic // hugeParam
```

Comment:
TODO: Should we just swallow all errors here? Or should we only swallow errors for unrecognized L1StandardBridge messages?

Reply:
Good point. Let's throw on all errors and add a TODO about needing to follow the spec in all cases.

Reply:
https://github.com/polymerdao/monomer/issues/234